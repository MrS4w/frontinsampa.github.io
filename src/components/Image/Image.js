import uuid from 'uuid/v4';
import React from 'react';
import PropTypes from 'prop-types';

const Image = ({
  src,
  alt,
  presentation,
  a11y,
}) => {
  const id = uuid();

  return presentation ? (
    /**
      * For decorative images like icons, illustrations and others.
      */
    <img src={src} srcSet={`${src} 2x`} alt="" role="presentation" />
  ) : (
    <React.Fragment>
      <img src={src} srcSet={`${src} 2x`} alt={alt} aria-describedby={id} />

      <span id={id}>
        { a11y.description }
      </span>
    </React.Fragment>
  );
};

Image.propTypes = {
  src: PropTypes.string.isRequired,
  alt: PropTypes.string.isRequired,
  presentation: PropTypes.bool,
  a11y: PropTypes.shape({
    title: PropTypes.string,
    description: PropTypes.string,
  }),
};

Image.defaultProps = {
  presentation: false,
  a11y: null,
};

export default Image;
