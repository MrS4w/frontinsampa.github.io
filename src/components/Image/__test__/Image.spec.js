import React from 'react';
import { shallow } from 'enzyme';

import Image from '../Image';

jest.mock('uuid/v4', () => jest.fn().mockImplementation(() => 'string'));

describe('Components: <Image />', () => {
  describe('With `presentation` as `true`.', () => {
    const wrapper = shallow(
      <Image src="/path/to/image.png" alt="" presentation />,
    );

    it('Create a snapshot.', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('With `presentation` props as `false`.', () => {
    const wrapper = shallow(
      <Image src="/path/to/image.png" alt="" a11y={{
        description: 'Lorem ipsum dolor sit amet consectetur adipisicing elit',
      }} />,
    );

    it('Create a snapshot.', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });
});
