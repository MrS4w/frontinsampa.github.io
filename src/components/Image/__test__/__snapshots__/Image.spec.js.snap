// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components: <Image /> With \`presentation\` as \`true\`. Create a snapshot. 1`] = `
<img
  alt=""
  role="presentation"
  src="/path/to/image.png"
  srcSet="/path/to/image.png 2x"
/>
`;

exports[`Components: <Image /> With \`presentation\` props as \`false\`. Create a snapshot. 1`] = `
<Fragment>
  <img
    alt=""
    aria-describedby="string"
    src="/path/to/image.png"
    srcSet="/path/to/image.png 2x"
  />
  <span
    id="string"
  >
    Lorem ipsum dolor sit amet consectetur adipisicing elit
  </span>
</Fragment>
`;
